content/jest.globalSetup.ts

```diff
@@ -0,0 +1,53 @@
+import { GenericContainer } from 'testcontainers'
+import { Container } from 'testcontainers/dist/container'
+import { LogWaitStrategy } from 'testcontainers/dist/wait-strategy'
+import { DEFAULT_DATABASE_CONFIG } from './src/Environment'
+import { E2ETestEnvironment } from './test/integration/E2ETestEnvironment'
+import { isCI } from './test/integration/E2ETestUtils'
+
+export default async (): Promise<void> => {
```

Comment:
After reading the whole PR, I'd like to suggest this: 

Wouldn't it be better to be more in control and use a high order function to test with a DB? These hooked files add a lot of magic and force us to have to separate unit and integration tests. Making coverage reports _very_ hard to unify.

```ts
export function describeWithDb(nameOfSuite: string, testFn: () => void) {
  describe(nameOfSuite, () => {
    it('initialize db', () => ...)
    testFn()
    it('teardown db', () => ...)
  })
}
```

With that function, the only thing needed would be to use `describeWithDb` instead of `describe`


```ts
describeWithDb('my test', () => {
  it('does something with the DB', () => ...)
})
```

Reply:
Otherwise I'd suggest to move it back to the `E2ETestEnvironment` to not scatter the logic around many files handled by runners, that makes the debugging and understanding of the tests flow hard to follow

Reply:
This change only applies to development env, preventing `E2ETestEnvironment` to spin up and delete a container every test, but not for initializing the DB. 
DB initialization is still happening in `E2ETestEnvironment` where it creates a database and run the migrations.

Also there is no problem to unify unit and integration tests, this change only ensure you have a postgres instance up and running, which is the same behavior as CircleCI having a service for postgres.

After this change:
- setup postgres instance
- run all tests (every test uses its own database)
- tear down

Before this change:
- for each test
  - start a container
  - run test (uses its own database)
  - delete container
This is causing a lot of noise when containers are not available or failing tests not deleting containers.

I agree if you are talking about removing `E2ETestEnvironment`, but it means a lot of refactor, probably worth to think it in another PR.

Reply:
I completely agree on unifying coverage reports. On that line, I introduced Jest `projects` [here](https://github.com/decentraland/catalyst/pull/847/commits/dfa9f433c764375ea07a7ccee60ab4858717e8db). We can have a single config file for both unit and integration tests and have specific configurations for each. If we run them together, it will produce a unified coverage report.

A PR for unifying content tests with parallelism on the CI is being developed here: https://github.com/decentraland/catalyst/pull/850